resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/ResourceTest.java

```diff
@@ -167,6 +167,26 @@ protected static boolean isWindows() {
 		return Platform.getOS().equals(Platform.OS_WIN32);
 	}
 
+	/**
+	 * Returns whether the current platform is linux.
+	 *
+	 * @return <code>true</code> if this platform is linux, and <code>false</code>
+	 *         otherwise.
+	 */
+	protected static boolean isLinux() {
+		return Platform.getOS().equals(Platform.OS_LINUX);
+	}
+
+	/**
+	 * Returns whether the current platform is mac OSX.
+	 *
+	 * @return <code>true</code> if this platform is mac OSX, and <code>false</code>
+	 *         otherwise.
+	 */
+	protected static boolean isMacOSX() {
+		return Platform.getOS().equals(Platform.OS_MACOSX);
+	}
+
```

Comment:
I'd propose to have these methods in a separate utility class (together with the pre-existing `isWindows` method) to stop fostering inheritance for reusing test utility methods by further messing up test base classes. Then these methods can also be used in other test classes nthat do not inheritfrom `ResourceTest`. Maybe this refactoring could be separated into an independent PR since it is unrelated to the addition of assumes/moving to JUnit 4 anyway.

Reply:
Sounds good, I already created an issue for it --> #448

Reply:
Why not add methods to the Platformclass directly? `Platform` already is some kind of utility class, so why we need a utility for a utility that the is used in another utility class?

Reply:
@laeubi  that could work too, thank you! I modified the issue #448 , in case you want to follow it :)

Reply:
See https://github.com/eclipse-platform/eclipse.platform/pull/704